sympy/sets/sets.py

```diff
@@ -994,6 +993,29 @@ def _eval_imageset(self, f):
             return
         var = f.variables[0]
 
+        if expr.is_Piecewise:
+            result = S.EmptySet
+            domain_set = self
+            for i in expr.args:
+                func = i[0]
+                if i[1] is S.true:
+                    intrvl = domain_set
+                else:
+                    intrvl = i[1].as_set()
+                    intrvl = Intersection(domain_set, intrvl)
+
+                if func.is_Number:
+                    image = FiniteSet(func)
+                else:
+                    image = imageset(Lambda(var, func), intrvl)
+                result = Union(result, image)
+
+                # remove the part which has been `imaged`
+                domain_set = Complement(domain_set, intrvl)
+                if domain_set.is_EmptySet:
+                    break
```

Comment:
Here `break` is used when the `self( or domain_set)` has been `exhausted`. `Return None` would `return None` instead of the `imaged_union( or result)`.